discovery/chan_series.go

```diff
@@ -204,11 +205,50 @@ func (c *ChanSeries) FilterKnownChanIDs(_ chainhash.Hash,
 	isZombieChan func(time.Time, time.Time) bool) (
 	[]lnwire.ShortChannelID, error) {
 
-	newChanIDs, err := c.graph.FilterKnownChanIDs(superSet, isZombieChan)
+	newChanIDs, knownZombies, err := c.graph.FilterKnownChanIDs(superSet)
 	if err != nil {
 		return nil, err
 	}
 
+	edgeLookup := make(map[uint64]graphdb.ChannelUpdateInfo)
+	for _, edge := range superSet {
+		edgeLookup[edge.ShortChannelID.ToUint64()] = edge
+	}
+
+	for _, scid := range knownZombies {
+		info, ok := edgeLookup[scid]
+		if !ok {
```

Comment:
I think this case should be impossible? It looks we can just [define an error out of existence](https://www.google.com/search?client=firefox-b-d&q=define+an+error+out+of+existence) by making `knownZombies` a slice of edge info?

Reply:
good idea!